sklearn/ensemble/gradient_boosting.py

```diff
@@ -1184,6 +1184,37 @@ def _validate_y(self, y):
         # Default implementation
         return y
 
+    def apply(self, X):
+        """Apply trees in the ensemble to X, return leaf indices.
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix, shape = [n_samples, n_features]
+            The input samples. Internally, it will be converted to
+            ``dtype=np.float32`` and if a sparse matrix is provided
+            to a sparse ``csr_matrix``.
+
+        Returns
+        -------
+        X_leaves : array_like, shape = [n_samples, n_estimators, n_classes]
```

Comment:
Ok I get it, removing the 3rd dimension for the regression case. I agree, we can override the apply method in `GradientBoostingRegressor` class to:

1- fix the docstring to avoid mentioning n_classes
2- do the ravelling.